### ISSUE TITLE
Playwright Utils: Add an easy way to switch between different user roles

### ISSUE THREAD

**Issue by Mamaduka** [WordPress/gutenberg#55751]
The current Playwright utils package needs a method for testing applications for different user roles. The new method should allow devs to switch between users with different permissions easily.

cc @kevin940726, @WunderBart, @swissspidy

**Comment by swissspidy** [comment]
Makes sense to me. I always found the Puppeteer-based utils for that a bit cumbersome to use and error-prone. But here with separate `storageState` files and `requestUtils` I think it's possible to build something nice.

**Comment by Mamaduka** [comment]
Playwright docs has some examples - https://playwright.dev/docs/auth#multiple-signed-in-roles
